### PR TITLE
🛡️ Security | Upgrade to latest dependencies in GHA & tweak installs/build

### PR DESCRIPTION
**Pull request by Fdawgs**
### Description

This PR:
- Bumps GitHub Actions to their latest major versions, which now use Node 16 internally instead of Node 12 (EOL as of end of April 2022, no longer receives security updates):
  - Changelog for `actions/checkout` can be [found here](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
  - Releases for `actions/setup-node` can be [found here](https://github.com/actions/setup-node/releases)

### References

N/A

### Testing

- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR or in Slab
- [ ] I have updated or added public documentation
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `main`

### CODE REVIEW

**Comment by Fdawgs** [comment]
There's probably a few more things that could be done to beef up CI security:
-   Declares the minimum permissions for CI workflows to run at the workflow level, following principle of least privilege; see [related GitHub security post](https://securitylab.github.com/research/github-actions-building-blocks/)
-   Removes Git credentials/SSH keys after checkout as a security precaution by setting `persist-credentials` to false for `actions/checkout` action

**Comment by zrosenbauer** [comment]
@Fdawgs good looking on the updating the actions, bad C+P on our end 😄 

I'll take a look at the permissions.

On the `npm install` that was intentional as the prismaclient is a peer dep, we have to rely on the installers dep due to prisma nesting the client in the `.prisma` dir unless I'm misunderstanding how prisma generates its client.

**Comment by zrosenbauer** [comment]
@Fdawgs looks like its still failing (no idea why tbh) also made changes to deploy process so might conflict

I'm happy to merge just the updates to the GHA actions and do the `npm` changes in diff PR?

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -14,13 +14,13 @@ jobs:
...
-        uses: actions/checkout@v1
+        uses: actions/checkout@v3
       - name: Use Node.js v18
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
         with:
           node-version: 18
       - name: Install Packages
-        run: npm install
+        run: npm i
```
Comment (zrosenbauer) [inline .github/workflows/ci.yaml:23]: This seems to work solves the `Module '"@prisma/client"' has no exported member 'PrismaClient'` error.

It looks like this is probably "fixing" it => https://github.com/prisma/prisma/blob/f718439571a60c91d3f5440226fe71031c84fa58/packages/client/package.json#L48

Maybe we `npm i @prisma/client & npx prisma generate`  and `npm i --ignore-scripts`?

Its only getting hit on the typecheck...

**Review by zrosenbauer — CHANGES_REQUESTED** [review]
Until we can figure out the `npm i` I don't want to merge, we can merge in the GHA upgrades though

**Comment by Fdawgs** [comment]
> @Fdawgs looks like its still failing (no idea why tbh) also made changes to deploy process so might conflict
> 
> I'm happy to merge just the updates to the GHA actions and do the `npm` changes in diff PR?

sounds good, will update PR.

**Review by zrosenbauer — APPROVED** [review]

**Pull request merged**